### ISSUE TITLE
The description of key_dedding_mask in float mode is incorrect

### ISSUE THREAD

**Issue by sealoongleft** [pytorch/pytorch#141018]
### 📚 The doc issue

The problematic [document](https://pytorch.org/docs/stable/generated/torch.nn.MultiheadAttention.html#torch.nn.MultiheadAttention) is about the description of key_padding_mask.
key_padding_mask：For a float mask, it will be directly added to the corresponding key value.

issue：I tested this float type and the results showed that it was not directly added to the key value, but did change the weight value in some form. The code and results are shown below.
```
import torch
import torch.nn as nn
import torch.nn.functional as F

multiheadattention=nn.MultiheadAttention(embed_dim=2,
num_heads=1,bias=False,batch_first=True)
data=torch.tensor([[[1,1.],[2,0.]]])
multiheadattention.in_proj_weight.data=torch.tensor(
[[1.,2],[0,1],[1,0],[1,1],[2,1],[2,0]])
multiheadattention.out_proj.weight.data=torch.tensor([[1,2],
[3,4.]])
keypaddingmask=torch.tensor([[1,2.]])
out1=multiheadattention(data,data,data,
need_weights=True,key_padding_mask=keypaddingmask)#When removing key_padding_mask, the returned weight tensor([[[0.1070, 0.8930], [0.1956, 0.8044]]] is correct, but adding key_padding_mask will result in an incorrect weight tensor([[[0.0422, 0.9578], [0.0821, 0.9179]]] as described in the document.
print(out1)
```

![image](https://github.com/user-attachments/assets/985e8af9-0a17-4f0e-af28-8f925c9b3258)

### Suggest a potential alternative/fix

Update the description about key_padding_mask.

**Comment by sealoongleft** [comment]
@joldov  
In fact, this is a very interesting question, but no one has answered me yet. I am patiently waiting.

**Comment by sealoongleft** [comment]
### I have asked this question many times and finally received a pull request hahaha🤣🤣🤣
